pymc3/distributions/multivariate.py

```diff
@@ -248,6 +249,50 @@ def _distr_parameters_for_repr(self):
         return ["mu", "cov"]
 
 
+class MvStudentTRV(RandomVariable):
+    name = "multivariate_studentt"
+    ndim_supp = 1
+    ndims_params = [0, 1, 2]
+    dtype = "floatX"
+    _print_name = ("MvStudentT", "\\operatorname{MvStudentT}")
+
+    def __call__(self, nu, mu=None, cov=None, size=None, **kwargs):
+
+        dtype = aesara.config.floatX if self.dtype == "floatX" else self.dtype
+
+        if mu is None:
+            mu = np.array([0.0], dtype=dtype)
+        if cov is None:
+            cov = np.array([[1.0]], dtype=dtype)
+        return super().__call__(nu, mu, cov, size=size, **kwargs)
+
+    def _shape_from_params(self, dist_params, rep_param_idx=1, param_shapes=None):
+        return default_shape_from_params(self.ndim_supp, dist_params, rep_param_idx, param_shapes)
+
+    @classmethod
+    def rng_fn(cls, rng, nu, mu, cov, size):
+
+        # Don't reassign broadcasted cov, since MvNormal expects two dimensional cov only.
+        mu, _ = broadcast_params([mu, cov], cls.ndims_params[1:])
+
+        chi2_samples = np.sqrt(rng.chisquare(nu, size=size) / nu)
+        # Add distribution shape to chi2 samples
+        chi2_samples = chi2_samples.reshape(chi2_samples.shape + (1,) * len(mu.shape))
+
+        mv_samples = pm.MvNormal.dist(
```

Comment:
This seems odd. Can't we call scipy/numpy MvNormal directly? Or just the rng classmethod of aesara's MvNormal op?

My understanding is that random ops should not depend on pymc distributions at all.

Reply:
> Can't we call scipy/numpy MvNormal directly

Unfortunately, scipy and numpy `multivariate_normal` generators cannot work when `mu` takes two-dimensional shape -

```python
>>> import numpy as np
>>> import scipy
>>> np.__version__, scipy.__version__
('1.20.3', '1.6.3')
>>> mu, cov = np.ones((10, 3)), np.eye(3)
>>> np.random.multivariate_normal(mean=mu, cov=cov)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "mtrand.pyx", line 4071, in numpy.random.mtrand.RandomState.multivariate_normal
ValueError: mean must be 1 dimensional
>>> from scipy.stats import multivariate_normal
>>> multivariate_normal.rvs(mean=mu, cov=cov)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/sayam/miniconda3/envs/pymc_dev/lib/python3.7/site-packages/scipy/stats/_multivariate.py", line 663, in rvs
    dim, mean, cov = self._process_parameters(None, mean, cov)
  File "/Users/sayam/miniconda3/envs/pymc_dev/lib/python3.7/site-packages/scipy/stats/_multivariate.py", line 413, in _process_parameters
    dim)
ValueError: Array 'mean' must be a vector of length 30.
>>> 
```

Aesara's [multivariate_normal RV](https://github.com/aesara-devs/aesara/blob/3acdd78e7a5d27e70cd8de3fc00bd5f010c65ee3/aesara/tensor/random/basic.py#L270-L288) handles all the cases. So that's why I am using `pm.MvNormal.dist` to access underneath implementation.

> just the rng classmethod of aesara's MvNormal op

Let me try this out